app/src/main/res/layout/fragment_posts.xml

```diff
@@ -1,36 +1,56 @@
 <?xml version="1.0" encoding="utf-8"?>
-<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     tools:context=".fragments.PostsFragment">
+
     <com.facebook.shimmer.ShimmerFrameLayout
         android:id="@+id/shimmerLayout"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content">
+        android:layout_width="406dp"
+        android:layout_height="669dp"
```

Comment:
Something about all these hardcoded widths and heights don't look right. They probably won't look as good on other devices with different screen sizes.

Normally you only need to define the margins/padding around the views and they'll auto-scale but maybe for these use cases you might need to look into defining percentages of parent layout that they should take up. You can look into this as a later stretch goal.